snaps/reference/permissions.md

```diff
@@ -179,3 +179,30 @@ Specify this permission in the manifest file as follows:
   "endowment:webassembly": {}
 },
 ```
+
+## Dynamic permissions
+
+Dynamic permissions are not requested in the manifest file. Instead, they are acquired during the lifecycle of the snap.
+
+## eth_accounts
+
+This dynamic permission is acquired when a snap calls the `eth_requestAccounts` method of the Ethereum provider. Calling `eth_requestAccounts` requires the [`ethereum-provider` endowment](#endowmentethereum-provider). The presence of the permission can be checked by calling [`wallet_getPermissions`](/wallet/reference/rpc-api/#wallet_getpermissions). If the permission is present, the return value of `wallet_getPermissions` will contain a permission with a `parentCapability` of `eth_accounts`. It will come with a caveat of `restrictReturnedAccounts`, an array of all the accounts that the user allowed for this snap:
```

Comment:
Worth mentioning that you should be able to request `eth_accounts` and/or `eth_requestAccounts`, which will also return the addresses without the extra permission caveats information?

Reply:
@FrederikBolding do you mean using `wallet_requestPermissions`? I didn't cover that in the PR, can't find any info on how to use it. My understanding is that calling `eth_accounts` won't popup the MetaMask window, so it won't generate any new permission. Can you elaborate on this please?

Reply:
Sorry, I mean after calling `eth_requestAccounts` to get the `eth_accounts` permission, you can simply call it again to get the already granted accounts. You don't need to use `wallet_getPermissions`. IIRC requesting `eth_accounts` should do the same. But fine by me to just reference `eth_requestAccounts`, it feels like a better pattern than `wallet_getPermissions`

Reply:
This is a technical reference that documents the permissions. A separate page explaining how to use `eth_requestAccounts` would provide a best practice.